fuel-vm/src/interpreter/initialization.rs

```diff
@@ -102,13 +102,13 @@ where
     /// For predicate estimation and verification, check [`Self::init_predicate`]
     pub fn init_script(
         &mut self,
-        checked: Checked<Tx>,
+        immutable: Ready<Tx>,
     ) -> Result<(), InterpreterError<S::DataError>> {
         let block_height = self.storage.block_height().map_err(RuntimeError::Storage)?;
 
         self.context = Context::Script { block_height };
 
-        let (tx, metadata): (Tx, Tx::Metadata) = checked.into();
+        let (_, tx, metadata, _) = immutable.decompose();
```

Comment:
You need to check that `gas_price` from `Read` is the same like in `Interpreter`. The same for the `deploy` method

Reply:
I think I addressed this. But maybe I misunderstood.